src/features/cart/CartProductCard.tsx

```diff
@@ -0,0 +1,58 @@
+import {CartItem, useCartStore} from "./cartStore.ts";
+import {useQuery} from "@tanstack/react-query";
+import {getProductWithId} from "../../services/api.ts";
+import {formattedPrice} from "../../utils/productUtils.ts";
+
+const CartProductCard = ({cartItem}: { cartItem: CartItem }) => {
+    const productQuery = useQuery({
+        queryKey: [cartItem.productId],
+        queryFn: () => getProductWithId(`${cartItem.productId}` ?? '')
+    })
+    const addItemToCart = useCartStore((state) => state.addItem);
```

Comment:
consider using derived state for the cart products, so that you don't make calls for each product

Reply:
I think there is a little confusion here with derived state 😄